nss-day-cohort-60/rare-server-billowing-smurfs
# Jl post crud

###Adds CRUD functionality to /posts request
Closes #23 
Closes #22 
Closes #20 
Closes #25 
Closes #19 
Closes #11 

##Additional Changes
In addition to the above tickets, I also altered the author property in Post model to Foreign Key instead of OneToOneField. New migrations are necessary before testing.

# To Test:
1. Pull branch jl-post-crud
2. GET request to http://localhost:8000/posts/1 for retrieve() method, confirm comments array is received
3. GET request to http://localhost:8000/posts for list() method, confirm comments array is received
4. POST request to http://localhost:8000/posts with the below body, then GET to confirm it posted.
{
    "category": 3,
    "title": "The Chuckling Mr. Munsel's Menagerie",
    "publication_date": "2019-10-03T10:45:34Z",
    "image_url": "https://cdn.britannica.com/36/123536-050-95CB0C6E/Variety-fruits-vegetables.jpg",
    "content": "Jumping jacks are for the jolly Jonah."
}
5. PUT request to http://localhost:8000/posts/3 and modify the body with the same properties as the above example.
6. DELETE request to  http://localhost:8000/posts/3, followed by a GET to confirm deletion.
7. Eat Cake.
